miden-tx/src/host/mod.rs

```diff
@@ -108,7 +108,7 @@ impl<A: AdviceProvider, T: TransactionAuthenticator> TransactionHost<A, T> {
     // EVENT HANDLERS
     // --------------------------------------------------------------------------------------------
 
-    fn on_note_created<S: ProcessState>(
+    fn on_note_after_created<S: ProcessState>(
         &mut self,
         process: &S,
     ) -> Result<(), TransactionKernelError> {
```

Comment:
Could we add a brief doc comment to this function? Also, let's move expected stack state from line 116 below into this doc comment.

In fact, let's do this for all event handlers here (i.e., put expected stack state into the doc comment).